controllers/api_keys.rb

```diff
@@ -0,0 +1,44 @@
+class CalendarEvents::ApiKeysController < ApplicationController
+
+  APPLICATION_NAME = 'discourse-events'
+  SCOPES = [CalendarEvents::USER_API_KEY_SCOPE]
+
+=begin
+  As soon as a new client_id is passed for the same API key, the key record
+  will be updated to contain the new client_id automatically.
+  See Auth::DefaultCurrentUserProvider#lookup_user_api_user_and_update_key
+
+  This means that rate limits could be exceeded in some cases.
+  TODO: Instead we should allow a unique key to be created for each client.
+=end
+  def index
```

Comment:
Prevent rails error when visiting this route without login. Show 404 preferably.

Reply:
Getting a redirect here, not an error:

![image](https://user-images.githubusercontent.com/6516575/64018209-130b0680-cb2c-11e9-930c-2e6161d14775.png)
![image](https://user-images.githubusercontent.com/6516575/64018214-19997e00-cb2c-11e9-8e1f-2758982631eb.png)

Reply:
Ah I see, it errors when `SiteSetting.login_required? == false` and nobody is logged in. Then it obviously doesn't know which user to give an API key to.